spline-shadow/rest-gateway/src/main/scala/za/co/absa/spline/gateway/rest/AppConfig.scala

```diff
@@ -16,9 +16,45 @@
 
 package za.co.absa.spline.gateway.rest
 
-import org.springframework.context.annotation.{Configuration, Import}
+import java.util.Arrays.asList
+
+import org.apache.commons.configuration.{CompositeConfiguration, EnvironmentConfiguration, SystemConfiguration}
+import org.springframework.context.annotation.{Bean, Configuration, Import}
+import org.springframework.web.filter.CorsFilter
+import org.springframework.web.cors.CorsConfiguration
+import org.springframework.web.cors.UrlBasedCorsConfigurationSource
+import za.co.absa.spline.common.config.ConfTyped
+import za.co.absa.spline.gateway.rest.CorsConfig.CorsFilterConf
 
 @Configuration
 @Import(Array(classOf[ArangoRepoConfig]))
-class AppConfig
+class AppConfig{
+
+
+
+  val source: UrlBasedCorsConfigurationSource  = new UrlBasedCorsConfigurationSource()
+  val config = new CorsConfiguration
+  config.setAllowCredentials(true)
+  if(CorsFilterConf.allowedOrigin != null) config.addAllowedOrigin(CorsFilterConf.allowedOrigin) else config.addAllowedOrigin("*")
+  CorsFilterConf.allowedHeader match {
+    case ls if ls.nonEmpty => CorsFilterConf.allowedHeader.foreach(config.addAllowedHeader)
+    case _ =>  config.addAllowedHeader("*")
+  }
+  config.addAllowedMethod("GET")
+  source.registerCorsConfiguration("/**", config)
+
+  @Bean def springFilterProxy = new CorsFilter(source)
+}
+
+
+object CorsConfig extends CompositeConfiguration(asList(
```

Comment:
`CorsConfig` should be places into a separate file. Usually (excluding some specific cases) in Scala only companion objects are placed in the same file with the corresponding class.